File: src/finanmind/services/usd_amount_parser.py
"""Parse USD amounts typed by users (digits, optional decimal dot)."""

import re


class UsdAmountParser:
    """Converts human-entered USD strings into floats."""

    _NON_DIGIT_DOT = re.compile(r"[^\d.]")

    @classmethod
    def parse(cls, raw: str) -> float:
        """Strip grouping commas and symbols; accept one decimal dot."""
        trimmed = raw.strip()
        if trimmed == "":
            raise ValueError("Vacío")
        cleaned = cls._NON_DIGIT_DOT.sub("", trimmed.replace(",", ""))
        if cleaned == "" or cleaned == ".":
            raise ValueError("Sin dígitos")
        if cleaned.count(".") > 1:
            raise ValueError("Use un solo punto decimal")
        return float(cleaned)

File: src/finanmind/webui/assets/js/format/usd_amount_parser.js
export class UsdAmountParser {
  /** Mirrors ``UsdAmountParser`` from Python: strip non-digit/non-dot, one dot max. */
  static parse(raw) {
    const trimmed = String(raw || "").trim();
    if (trimmed === "") throw new Error("Vacío");
    const noCommas = trimmed.replace(/,/g, "");
    const cleaned = noCommas.replace(/[^\d.]/g, "");
    if (cleaned === "" || cleaned === ".") throw new Error("Sin dígitos");
    if (cleaned.split(".").length > 2) throw new Error("Use un solo punto decimal");
    return Number(cleaned);
  }
}
